modules/posts/includes/class-post-view-base.php

```diff
@@ -23,4 +46,721 @@ function post() {
+		));
+		$args[ 'previous' ] = false;
+		return $this->get_adjacent_post_link( $args );
+
+	}
+
+	/**
+	 * @param array $args
+	 *
+	 * @return mixed|null
+	 */
+	function the_previous_post_link( $args = array() ) {
+
+		echo $this->get_previous_post_link( $args );
+
+	}
+
+	/**
+	 * @param array $args
+	 *
+	 * @return mixed|null
+	 */
+	function the_next_post_link( $args = array() ) {
+
+		echo $this->get_next_post_link( $args );
+
+	}
+
+	function the_css_classes_attr( $classes = false ) {
+
+		echo $this->get_css_classes_attr( $classes );
+
+	}
+
+	function get_css_classes_attr( $classes = false ) {
+
+		return get_post_class( $classes, $this->model()->ID() );
+
+	}
+
+	/**
+	 * @param array $args
+	 * @return string
+	 */
+	function the_multipage_links_html( $args = array() ) {
+
+		echo wp_kses_post( $this->get_multipage_links_html( $args ) );
+
+	}
+
+	/**
+	 * @param array $args
+	 * @return string
+	 */
+	function get_multipage_links_html( $args = array() ) {
+
+		$args = wp_parse_args( $args, array(
+
+			'before' => '<div class="page-links">' . esc_html__( 'Pages:', 'wplib' ),
+			'after'  => '</div>',
+
+		));
+
+		$args[ 'echo' ] = false;
+
+		$globals = $this->_save_multipage_globals();
+
+		if ( false !== $this->multipage ) {
+
+			$this->_set_multipage_globals();
+
+		}
+
+		$links_html = wp_link_pages( $args );
+
+		$this->_restore_multipage_globals( $globals );
+
+		return $links_html;
+
+	}
+
+	/**
+	 * Set $this->multipage property from the multipage global variables
+	 */
+	private function _set_multipage_property() {
+
+		global $page, $numpages, $multipage, $more;
+		if ( $multipage ) {
+
+			$this->multipage = (object) array(
+				'page'     => $page,
+				'numpages' => $numpages,
+				'more'     => $more,
+			);
+
+		}
+
+	}
+
+	/**
+	 * Set multipage global variables from $this->multipage
+	 */
+	private function _set_multipage_globals() {
+
+		global $multipage, $page, $numpages, $more;
+
+		$multipage = true;
+		$page      = $this->multipage->page;
+		$numpages  = $this->multipage->numpages;
+		$more      = $this->multipage->more;
+
+	}
+
+	/**
+	 * @return array
+	 */
+	private function _save_multipage_globals() {
+
+		global $page, $numpages, $multipage, $more;
+		return compact( $page, $numpages, $multipage, $more );
+
+	}
+
+	/**
+	 * @param array $globals
+	 */
+	private function _restore_multipage_globals( $globals ) {
+
+		global $page, $numpages, $multipage, $more;
+		call_user_func( 'extract', $globals );
+
+	}
+
+	/**
+	 * @param bool $method_name
+	 * @param string $to_output
+	 */
+	function the( $method_name, $to_output ) {
+
+		if ( method_exists( $this->model, $method_name ) && is_callable( $callable = array( $this->model, $method_name ) ) ) {
+
+			if ( call_user_func( $callable ) ) {
+
+				echo $to_output;
+
+			}
+
+		}
+
+
+	}
+
+	/**
+	 * @param bool $method_name
+	 * @param string $to_output
+	 */
+	function the_not( $method_name, $to_output ) {
```

Comment:
It allows a method name on the model to be used as a binary choice for not outputting something, i.e. 

```<div class="post <?php $object->the_not( 'is_single', 'not-single' ); ?>">```

I was trying to get around having that logic appear in each every place we needed to conditionally output or not output something.  It's inverse is `$object->the()` above.  

I don't love it, do you have a better idea?  Maybe we could overload `the()` for the not case like this?

```<div class="post <?php $object->the( '!is_single', 'not-single' ); ?>">```